internal/app/orchestrator/downstream.go

```diff
@@ -14,82 +14,73 @@ package orchestrator
 import (
 	"sync"
 
-	"github.com/envoyproxy/xds-relay/internal/app/mapper"
 	"github.com/envoyproxy/xds-relay/internal/app/transport"
 )
 
 // downstreamResponseMap is a map of downstream xDS client requests to response
 // channels.
 type downstreamResponseMap struct {
 	mu      sync.RWMutex
-	watches map[transport.Request]transport.Watch
+	watches map[string][]transport.Watch
 }
 
 func newDownstreamResponseMap() downstreamResponseMap {
 	return downstreamResponseMap{
-		watches: make(map[transport.Request]transport.Watch),
+		watches: make(map[string][]transport.Watch),
 	}
 }
 
 // createWatch initializes a new channel for a request if it doesn't already
 // exist.
-func (d *downstreamResponseMap) createWatch(req transport.Request) transport.Watch {
+func (d *downstreamResponseMap) addWatch(aggregatedKey string, w transport.Watch) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
-	if _, ok := d.watches[req]; !ok {
-		d.watches[req] = req.CreateWatch()
+	if _, ok := d.watches[aggregatedKey]; !ok {
+		d.watches[aggregatedKey] = make([]transport.Watch, 0, 10)
```

Comment:
What's the reasoning behind the 10 buffer choice here?

Reply:
it can be 1. I was experimenting with different initial size allocations to see their impact on benchmark